Integrations/integration-HPE_Service_Manager.yml

```diff
@@ -0,0 +1,489 @@
+configuration:
+- display: Server URL (e.g. https://192.168.0.1:13080)
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Username
+  name: username
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    var url = params.url + '/SM/9/rest';
+
+    // handle '/' at the end of the url
+    if (url[url.length - 1] === '/') {
+        url = url.substring(0, url.length - 1);
+    }
+
+    // var token = 'U3lzdGVtLkFkbWluOg==';
+    var token = Base64.encode(params.username.identifier + ':' + params.username.password);
+    // throw Base64.encode(params.username.identifier + ':' + params.username.password);
+
+    var result;
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            result = 'ok';
+            break;
+        case 'hpsm-create-incident':
+            result = createIncident();
+            break;
+        case 'hpsm-list-incidents':
+            result = listIncidents();
+            break;
+        case 'hpsm-get-incident-by-id':
+            result = getIncidentById();
+            break;
+        case 'hpsm-list-devices':
+            result = listDevices();
+            break;
+        case 'hpsm-get-device':
+            result = getDeviceById();
+            break;
+        default:
+            // You can use args[argName] or args.argName to get a specific arg. args are strings.
+            // You can use params[paramName] or params.paramName to get a specific params.
+            // Params are of the type given in the integration page creation.
+    }
+    return result;
+
+    function createIncident() {
+        var newIncident = {
+            Incident: {
+                Category: args.category,
+                Description: [
+                    args.description
+                ],
+                Service: args.service,
+                Title: args.title,
+            }
+        };
+        if (args.impact) {
```

Comment:
you can make a loop here that will fill the fields with exist...
an array with the field names, and if an arg exist add it
you can also use value ? value : undefined so this can be done in 3 lines

Reply:
`field_name: undefined` will still create field with such name.
And create list/map of prop names is less readable than doing this in explicit way
```
        argsExist = {
            impact: 'Impact',
            urgency: 'Urgency',
            alertStatus: 'AlertStatus',
            .....
        };

        for (var prop in argsExist) {
            if (argsExist.hasOwnProperty(prop)) {
                var propName = argsExist[prop];
                newIncident.Incident[propName] = args[prop];   
            }
        }
```